stdlib/cmake/modules/AddSwiftStdlib.cmake

```diff
@@ -2651,39 +2689,158 @@ function(add_swift_target_executable name)
     message(SEND_ERROR "${name} is using EXCLUDE_FROM_ALL which is deprecated.")
   endif()
 
+  if("${SWIFTEXE_TARGET_INSTALL_IN_COMPONENT}" STREQUAL "")
```

Comment:
I don't see this variable mentioned anywhere else in this repository. If this is its only use, maybe it's an appropriate place to add a comment that clarify what this variable is for and how it's used?

Reply:
It's one of the arguments to the `add_swift_target_executable` function; it gets set by the argument parsing at the top to equal the `INSTALL_IN_COMPONENT` setting.

Reply:
Is the name of the variable "assembled" dynamically somehow through string interpolation? If so, is that the reason there are no more references to it in code anywhere else by its full name then?

Reply:
It's done by `cmake_parse_arguments` at the top of the function. That function takes the prefix to prepend and then three lists of options as well as the actual arguments you want to parse.